nicsdru_nidirect_theme.theme

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @file
+ * Functions to support theming in the nicsdru_nidirect_theme theme.
+ */
+
+use Drupal\user\Entity\User;
+
+/**
+ * Implements hook_preprocess_field().
+ */
+function nicsdru_nidirect_theme_preprocess_field(&$variables) {
+  // Implement audit link.
+  if (isset($variables['element'])
+    && isset($variables['element']['#entity_type'])
+    && ($variables['element']['#entity_type'] == 'node')
+  ) {
+    // Only start this processing if the 'nidirect_workflow'
+    // module is installed.
+    if (\Drupal::service('module_handler')->moduleExists('nidirect_workflow')) {
+      $content_type = $variables['element']['#bundle'];
+      switch ($content_type) {
+        case 'health_condition':
+          // For health conditions, just send the user to the node edit
+          // form, where they can set the 'next review date' to a date
+          // in the future to remove this node from the 'needs audit' view.
+          $node = \Drupal::routeMatch()->getParameter('node');
+          if (!empty($node) && is_object($node)) {
```

Comment:
I don't think you need the `is_object()` check as it'll either be a node object from the routematch service, or null which satisfies the first condition.

Do double check the return types on the service; if so: that can be simplified a little.

Reply:
Well, yes, have checked the return values and it is either null or 'mixed'. So in that case I could get away with a simple:

if ($node)

but I feel slightly uncomfortable performing an operation like '$node->id()' on something that I haven't fully checked as the fallout can be nasty as you know.